### PR TITLE
[REF] web,*: clean legacy and module dependencies

### PR DESCRIPTION
**Pull request by aab-odoo**
Task 4119889
Task 3446824
Task 3586757

### CODE REVIEW

**Comment by aab-odoo** [comment]
robodoo rebase-ff

File: addons/web/static/tests/modules/dependencies.test.js [hunk 1]
```diff
@@ -30,11 +30,10 @@ function invalidImportsFrom(folder, allowedFolders) {
 
 describe.current.tags("headless");
 
-test("modules only import from allowed folders", () => {
+test.only("modules only import from allowed folders", () => {
```
Comment (Polymorphe57) [inline addons/web/static/tests/modules/dependencies.test.js]: remove

File: addons/web/static/src/views/debug_items.js [hunk 2]
```diff
@@ -85,14 +79,7 @@ export function editSearchView({ accessRights, component, env }) {
...
     }
-    let { searchViewId } = component.props.info || {}; // fallback is there for legacy
-    if ("viewParams" in component.props) {
-        //legacy
-        if (!component.props.viewParams.action.controlPanelFieldsView) {
-            return null;
-        }
-        searchViewId = component.props.viewParams.action.controlPanelFieldsView.view_id;
-    }
+    const { searchViewId } = component.componentProps.info;
```
Comment (Polymorphe57) [inline addons/web/static/src/views/debug_items.js:82]: component.props.info?

**Comment by Polymorphe57** [comment]
@robodoo r+

**Pull request closed without merging**